pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -322,41 +362,75 @@ func groupByAttributeKeyTags(keys map[string]v3.AttributeKey, tags ...v3.Attribu
 }
 
 // orderBy returns a string of comma separated tags for order by clause
+// if there are remaining items which are not present in tags they are also added
 // if the order is not specified, it defaults to ASC
-func orderBy(items []v3.OrderBy, tags []string) string {
+func orderBy(panelType v3.PanelType, items []v3.OrderBy, tags []string, keys map[string]v3.AttributeKey) []string {
 	var orderBy []string
+
+	// create a lookup
+	addedToOrderBy := map[string]bool{}
+	itemsLookup := map[string]v3.OrderBy{}
+
+	for i := 0; i < len(items); i++ {
+		addedToOrderBy[items[i].ColumnName] = false
+		itemsLookup[items[i].ColumnName] = items[i]
+	}
+
 	for _, tag := range tags {
-		found := false
-		for _, item := range items {
-			if item.ColumnName == tag {
-				found = true
-				orderBy = append(orderBy, fmt.Sprintf("%s %s", item.ColumnName, item.Order))
-				break
-			}
-		}
-		if !found {
+		if item, ok := itemsLookup[tag]; ok {
+			orderBy = append(orderBy, fmt.Sprintf("%s %s", item.ColumnName, item.Order))
+			addedToOrderBy[item.ColumnName] = true
+		} else {
 			orderBy = append(orderBy, fmt.Sprintf("%s ASC", tag))
 		}
 	}
 
-	// users might want to order by value of aggreagation
+	// users might want to order by value of aggregation
 	for _, item := range items {
 		if item.ColumnName == constants.SigNozOrderByValue {
 			orderBy = append(orderBy, fmt.Sprintf("value %s", item.Order))
+			addedToOrderBy[item.ColumnName] = true
+		}
+	}
+
+	// add the remaining items
+	if panelType == v3.PanelTypeList {
+		for _, item := range items {
+			// since these are not present in tags we will have to select them correctly
+			// for list view there is no need to check if it was added since they wont be added yet but this is just for safety
+			if !addedToOrderBy[item.ColumnName] {
+				attr := v3.AttributeKey{Key: item.ColumnName, DataType: item.DataType, Type: item.Type, IsColumn: item.IsColumn}
+				name := getColumnName(attr, keys)
+				orderBy = append(orderBy, fmt.Sprintf("%s %s", name, item.Order))
+			}
 		}
 	}
-	return strings.Join(orderBy, ",")
+	return orderBy
 }
 
-func orderByAttributeKeyTags(items []v3.OrderBy, tags []v3.AttributeKey) string {
+func orderByAttributeKeyTags(panelType v3.PanelType, aggregatorOperator v3.AggregateOperator, items []v3.OrderBy, tags []v3.AttributeKey, keys map[string]v3.AttributeKey) string {
 	var groupTags []string
 	for _, tag := range tags {
 		groupTags = append(groupTags, tag.Key)
 	}
-	str := orderBy(items, groupTags)
-	if len(str) > 0 {
-		str = str + ","
+	orderByArray := orderBy(panelType, items, groupTags, keys)
+
+	found := false
+	for i := 0; i < len(orderByArray); i++ {
+		if strings.Compare(orderByArray[i], constants.TIMESTAMP) == 0 {
+			orderByArray[i] = "ts"
+			break
+		}
```

Comment:
What is happening here? 
- The result `orderByArray` will have order asc/desc, so it will never match the `constants.TIMESTAMP`?
- Why does it replace with simple `ts`? Is this the correct alias for all cases?